signwriting_transcription/pose_to_signwriting/joeynmt_pose/tokenizer.py

```diff
@@ -174,10 +206,22 @@ def __init__(
                   'eos_token': EOS_TOKEN,
                   'pad_token': PAD_TOKEN,
                   'unk_token': UNK_TOKEN}
+        self.shared_state = shared_state
         self.tokenizer = SignWritingTokenizer(starting_index=None, **kwargs)
 
     def __call__(self, raw_input: str, is_train: bool = False) -> List[str]:
         """Tokenize"""
+        if raw_input == "SYNTHETIC":
+            if "generated_pose" not in self.shared_state.data.keys():
+                raw_input = ""
+            else:
+                generated_pose = self.shared_state.get("generated_pose")
+                if generated_pose is None:
+                    raise ValueError("Generated pose is not found in shared state.")
+                generated_right_hand = str(hex(generated_pose.keyframes[0].left_hand_signwriting))[2:]
+                generated_left_hand = str(hex(generated_pose.keyframes[0].right_hand_signwriting))[2:]
+                generated_fsw = f'M538x552S2ff00482x483S{generated_right_hand}522x522S{generated_left_hand}455x521'
```

Comment:
This function should belong in the synthetic library, not here. Also, I see that you don't have support for face.

Reply:
I added new push request in the synthetic signwiritng, its just one function is fill free if you think that it need to be written differently to tell me, and I will try to do the modification.